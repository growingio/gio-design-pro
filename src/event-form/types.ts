export interface DeviceInfo {
  name: string;
  domain: string;
  os: string;
  // 以下是原生app特有的
  urlScheme?: string;
  projectId: string;
  screenHeight: number;
  screenWidth: number;
  sdkVersion: string;
  sdkVersionCode: number;
  timestamp: number;
}
export enum AppType {
  MINP,
  NATIVE,
  WEB,
}
export interface Rect {
  top?: number;
  left?: number;
  width?: number;
  height?: number;
  // 是不是在可视区外边
  inOut?: boolean;
}
