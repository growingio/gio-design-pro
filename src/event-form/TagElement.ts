// import { Element } from '../types';

// export declare type TagElement = Element;
export interface TagElement {
  id: string;
  name: string;
  actions: string[];
  attrs: DocProps;
  definition: DocProps;
  screenshot: Screenshot;
  platforms: string[];
  docType: string;
  description: string;
  businessType: string;
  isSystem: Boolean;
  createdAt: string;
  updatedAt: string;
  creatorId: string;
  updaterId: string;
  creator: string;
  updater: string;
  patternMatched: Boolean;
  appVersion: string;
  sdkVersion: string;
}

export interface DocProps {
  urlScheme?: string;
  domain?: string;
  path?: string;
  query?: string;
  xpath?: string;
  index?: string;
  href?: string;
  content?: string;
  pg?: string;
  contentType?: string; // 'match_phrase' | '=';
}
export interface Screenshot {
  target?: string;
  viewport: string;
}
