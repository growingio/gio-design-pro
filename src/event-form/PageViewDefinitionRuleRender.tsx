/* eslint-disable react/no-danger */
import React from 'react';
import { DocProps, TagElement } from './TagElement';
import { AppType } from './types';

const renderDefinitionRuleText = (props: { definition: DocProps; repeatTag?: TagElement; appType: AppType }) => {
  const { definition, repeatTag, appType } = props;
  let text = '现在定义的是';
  if (repeatTag) {
    text = `该规则已被 <b>${repeatTag.creator}</b> 定义为【${repeatTag.name}】。${text}`;
  }

  if (appType === AppType.NATIVE) {
    text += `页面 <span class="link">${definition.path}</span> 。`;
  } else if (appType === AppType.WEB) {
    if (definition.path) {
      text += `页面 <span class="link">${definition.domain + definition.path}</span> `;
    } else {
      text += ` <span class="link">${definition.domain}</span> `;
    }
    if (definition.query) {
      if (definition.path) text += '，';
      text += `查询条件为 <span class="link">${definition.query}</span>`;
    }
    if (definition.path) {
      text += ' 。';
    } else {
      text += ' 下的所有页面。';
    }
  } else if (appType === AppType.MINP) {
    if (definition.path === undefined) {
      text += ` <span class="link">小程序所有页面</span> `;
    } else {
      text += `页面 <span class="link">${definition.path}</span> `;
    }
    if (definition.query) {
      text += `，查询条件为 <span class="link">${definition.query}</span>`;
    }
    text += '。';
  }
  return <span dangerouslySetInnerHTML={{ __html: text }} />;
};
export default renderDefinitionRuleText;
