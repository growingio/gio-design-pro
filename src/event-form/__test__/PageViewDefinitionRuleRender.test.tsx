import React from 'react';
// import { fireEvent, render, screen } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import PageViewDefinitionRuleRender from '../PageViewDefinitionRuleRender';
import { TagElement } from '../TagElement';
import { AppType } from '../types';

describe('PageViewDefinitionRuleRender', () => {
  it('render AppType.WEB', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.WEB} />);
    expect(container).toMatchSnapshot();
    expect(screen.queryByText(/现在定义的是页面/)).toBeTruthy();
  });
  it('render AppType.WEB and path==undefined', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      // path: 'pages/index/index',
      query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.WEB} />);
    expect(container).toMatchSnapshot();
  });
  it('render AppType.WEB and query==undefined', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      // query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.WEB} />);
    expect(container).toMatchSnapshot();
  });
  it('render AppType.MINP', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.MINP} />);
    expect(container).toMatchSnapshot();
    expect(screen.queryByText(/现在定义的是页面/)).toBeTruthy();
  });
  it('render AppType.MINP and path==undefined', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      // path: 'pages/index/index',
      query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.MINP} />);
    expect(container).toMatchSnapshot();
    expect(screen.queryByText(/小程序所有页面/)).toBeTruthy();
  });
  it('render AppType.MINP and query==undefined', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      // query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.MINP} />);
    expect(container).toMatchSnapshot();
  });
  it('render AppType.NATIVE', async () => {
    const definition = {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: 'a=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      // pg: null,
      contentType: '=',
      // urlScheme: null,
      __typename: 'DocProps',
    };
    const { container } = render(<PageViewDefinitionRuleRender definition={definition} appType={AppType.NATIVE} />);
    expect(container).toMatchSnapshot();
    expect(screen.queryByText(/现在定义的是页面/)).toBeTruthy();
  });
  it('render with repeatTag', async () => {
    const definition = {
      domain: 'release-messages.growingio.cn',
      path: 'pages/index/index',
      // query: undefined,
      // xpath: '/div.title/form.basic-grey/h1/span',
      index: '',
      href: '/link',
      content: '元素内容',
    };

    const repeatTag = {
      id: 'kqQe1qQr',
      name: '定义一个新的事件2',
      creator: 'zhangxueqi',
      creatorId: 'oNGzxQgB',
      createdAt: '2021-01-08T06:21:14Z',
      updater: 'zhangxueqi',
      updaterId: 'oNGzxQgB',
      updatedAt: '2021-01-08T06:21:14Z',
      description: '',
      platforms: ['minp'],
      docType: 'elem',
      actions: ['clck'],
      patternMatched: false,
      attrs: {
        content: '主页',
        index: '0',
        href: 'pages/index/index',
        path: 'pages/index/index',
        // query: null,
        xpath: '#onTabItemTap',
        // contentType: null,
        domain: 'wx265d0fa6fa70fae9',
        // urlScheme: null,
        __typename: 'DocProps',
      },
      definition: {
        domain: 'wx265d0fa6fa70fae9',
        path: 'pages/index/index',
        query: '12=12',
        xpath: '#onTabItemTap',
        index: '0',
        href: 'pages/index/index',
        content: '主页',
        // pg: null,
        contentType: '=',
        // urlScheme: null,
        __typename: 'DocProps',
      },
      screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    };
    const { container } = render(
      <PageViewDefinitionRuleRender
        repeatTag={(repeatTag as unknown) as TagElement}
        definition={definition}
        appType={AppType.MINP}
      />
    );
    expect(container).toMatchSnapshot();

    expect(screen.findByText(/^该规则已被/)).toBeTruthy();
  });
});
