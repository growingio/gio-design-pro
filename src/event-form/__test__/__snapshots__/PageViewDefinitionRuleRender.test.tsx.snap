// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageViewDefinitionRuleRender render AppType.MINP 1`] = `
<div>
  <span>
    现在定义的是页面 
    <span
      class="link"
    >
      pages/index/index
    </span>
     ，查询条件为 
    <span
      class="link"
    >
      a=12
    </span>
    。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.MINP and path==undefined 1`] = `
<div>
  <span>
    现在定义的是 
    <span
      class="link"
    >
      小程序所有页面
    </span>
     ，查询条件为 
    <span
      class="link"
    >
      a=12
    </span>
    。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.MINP and query==undefined 1`] = `
<div>
  <span>
    现在定义的是页面 
    <span
      class="link"
    >
      pages/index/index
    </span>
     。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.NATIVE 1`] = `
<div>
  <span>
    现在定义的是页面 
    <span
      class="link"
    >
      pages/index/index
    </span>
     。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.WEB 1`] = `
<div>
  <span>
    现在定义的是页面 
    <span
      class="link"
    >
      wx265d0fa6fa70fae9pages/index/index
    </span>
     ，查询条件为 
    <span
      class="link"
    >
      a=12
    </span>
     。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.WEB and path==undefined 1`] = `
<div>
  <span>
    现在定义的是 
    <span
      class="link"
    >
      wx265d0fa6fa70fae9
    </span>
     查询条件为 
    <span
      class="link"
    >
      a=12
    </span>
     下的所有页面。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render AppType.WEB and query==undefined 1`] = `
<div>
  <span>
    现在定义的是页面 
    <span
      class="link"
    >
      wx265d0fa6fa70fae9pages/index/index
    </span>
      。
  </span>
</div>
`;

exports[`PageViewDefinitionRuleRender render with repeatTag 1`] = `
<div>
  <span>
    该规则已被 
    <b>
      zhangxueqi
    </b>
     定义为【定义一个新的事件2】。现在定义的是页面 
    <span
      class="link"
    >
      pages/index/index
    </span>
     。
  </span>
</div>
`;
